collector/cpu_linux.go

```diff
@@ -83,8 +114,64 @@ func NewStatCollector() (Collector, error) {
 	}, nil
 }
 
+func (c *cpuCollector) Update(ch chan<- prometheus.Metric) error {
+	err1 := c.UpdateStat(ch)
+	err2 := c.UpdateCpufreq(ch)
+	if err1 != nil {
+		return err1
+	}
+	if err2 != nil {
+		return err2
+	}
```

Comment:
Maybe just use the same err variable?
```
err := c.UpdateStat(ch)
err = c.UpdateCpufreq(ch)
if err ... 
```
Not sure if this is ideal but imo slightly better.